pxemgr/etcd_discovery_handlers.go

```diff
@@ -0,0 +1,139 @@
+package pxemgr
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/gorilla/mux"
+)
+
+type EtcdNode struct {
+	Key   string      `json:"key"`
+	Value string      `json:"value,omitempty"`
+	Nodes []*EtcdNode `json:"nodes,omitempty"`
+	Dir   bool        `json:"dir,omitempty"`
+}
+
+type EtcdResponse struct {
+	Action string    `json:"action"`
+	Node   *EtcdNode `json:"node,omitempty"`
+}
+
+type EtcdResponseError struct {
+	ErrorCode int    `json:"errorCode"`
+	Message   string `json:"message"`
+	Cause     string `json:"cause"`
+}
+
+func (mgr *pxeManagerT) defineEtcdDiscoveryRoutes(etcdRouter *mux.Router) {
+	etcdRouter.PathPrefix("/new").Methods("PUT").HandlerFunc(mgr.etcdDiscoveryNewCluster)
+
+	tokenRouter := etcdRouter.PathPrefix("/{token:[a-f0-9]{32}}").Subrouter()
+	tokenRouter.PathPrefix("/_config/size").Methods("GET").HandlerFunc(mgr.etcdDiscoveryProxyHandler)
+	tokenRouter.PathPrefix("/_config/size").Methods("PUT").HandlerFunc(mgr.etcdDiscoveryProxyHandler)
+	tokenRouter.PathPrefix("/{machine}").Methods("PUT").HandlerFunc(mgr.etcdDiscoveryProxyHandler)
+	tokenRouter.PathPrefix("/{machine}").Methods("GET").HandlerFunc(mgr.etcdDiscoveryProxyHandler)
+	tokenRouter.PathPrefix("/{machine}").Methods("DELETE").HandlerFunc(mgr.etcdDiscoveryProxyHandler)
+	tokenRouter.Methods("GET").HandlerFunc(mgr.etcdDiscoveryProxyHandler)
+
+	etcdRouter.Methods("GET").HandlerFunc(mgr.etcdDiscoveryHandler)
+}
+
+func (mgr *pxeManagerT) etcdDiscoveryHandler(w http.ResponseWriter, r *http.Request) {
+	http.Redirect(w, r,
+		"https://github.com/giantswarm/mayu/blob/master/docs/etcd-discovery.md",
+		http.StatusMovedPermanently,
+	)
+}
+
+func (mgr *pxeManagerT) etcdDiscoveryNewCluster(w http.ResponseWriter, r *http.Request) {
+	var err error
+	size := mgr.defaultEtcdQuorumSize
+	s := r.FormValue("size")
+	if s != "" {
```

Comment:
yes it is fine if size is emptry. then the default size is used. check 2 lines above.